sources/ghostdog/src/main.rs

```diff
@@ -125,6 +241,46 @@ fn parse_device_name(device_info: &[u8], path: String) -> Result<String> {
         .to_string())
 }
 
+/// Read a file into a SupportedDevicesConfiguration Enum
+fn read_supported_devices_file(path: PathBuf) -> Result<SupportedDevicesConfiguration> {
+    let mut supported_devices_file =
+        fs::File::open(&path).context(error::OpenFileSnafu { path: path.clone() })?;
+    let mut supported_devices_str = String::new();
+    supported_devices_file
+        .read_to_string(&mut supported_devices_str)
+        .context(error::ReadFileSnafu { path: path.clone() })?;
+    let device_configuration: SupportedDevicesConfiguration =
+        serde_json::from_str(supported_devices_str.as_str())
+            .context(error::ParseGpuDevicesFileSnafu {})?;
+    Ok(device_configuration)
+}
+
+/// Given a PCI ID, search the Open GPU Supported Devices File to determine if the Open GPU Driver should be used
+fn find_preferred_driver(pci_id: String) -> Result<String> {
+    // This corresponds to T4 devices which are Turing based and supported, but currently do not show up in the supported-devices file
+    if pci_id == "10DE:1EB8" {
+        return Ok("nvidia-open-gpu".to_string());
+    }
+    let open_gpu_devices = read_supported_devices_file(OPEN_GPU_SUPPORTED_DEVICES_PATH.into())?;
+    let input_device =
+        PciDevice::from_str(pci_id.as_str()).map_err(|message| error::Error::ParsePciId {
+            pci_id: pci_id.clone(),
+            message,
+        })?;
+    match open_gpu_devices {
+        SupportedDevicesConfiguration::OpenGpu(device_list) => {
+            for supported_device in device_list.iter() {
+                if supported_device.device_id == format!("0x{}", input_device.device_id) {
```

Comment:
nit: perhaps move the format! out of the inner loop? The loop does not iterate over terribly many instances, so not a big deal.